src/blocked.c

```diff
@@ -287,9 +288,14 @@ void disconnectAllBlockedClients(void) {
              * which the command is already in progress in a way. */
             if (c->bstate.btype == BLOCKED_POSTPONE) continue;
 
-            unblockClientOnError(c, "-UNBLOCKED force unblock from blocking operation, "
-                                    "instance state changed (master -> replica?)");
-            c->flag.close_after_reply = 1;
+            if (!server.cluster_enabled && c->capa & CLIENT_CAPA_REDIRECT && server.primary_host) {
```

Comment:
I realized this is too simplistic, we need to look into the blocking type in order to find out whether redirecting is the correct response. I will update this part.

Reply:
done